lib/engine/game/g_1862/step/merge.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G1862
+      module Step
+        class Merge < Engine::Step::Base
+          def actions(entity)
+            return [] if !entity.corporation? || entity != current_entity
+            return [] if entity.receivership? || @game.skip_round[entity]
+
+            return ['choose'] if @merging
+
+            %w[merge pass]
+          end
+
+          def log_skip(entity)
+            super unless @game.skip_round[entity]
+          end
+
+          def auto_actions(entity)
+            return super if @merging
+
+            return [Engine::Action::Pass.new(entity)] if mergeable_candidates(entity).empty?
+
+            super
+          end
+
+          def merge_name
+            'Merge'
+          end
+
+          def merger_auto_pass_entity
+            current_entity
+          end
+
+          def can_merge?(entity)
+            mergeable_candidates(entity).any?
+          end
+
+          def description
+            return 'Choose Survivor' if @merging
+
+            'Merge with Major Corporation'
+          end
+
+          def process_merge(action)
+            @merging = [action.entity, action.corporation]
+            @log << "#{@merging.first.name} and #{@merging.last.name} will merge"
+          end
+
+          def process_choose(action)
+            if action.choice == :first
+              @log << "#{@merging.last.name} (non-survivor) will merge into #{@merging.first.name} (survivor)"
+              survivor = @merging.first
+              nonsurvivor = @merging.last
+            else
+              @log << "#{@merging.first.name} (non-survivor) will merge into #{@merging.last.name} (survivor)"
+              survivor = @merging.last
+              nonsurvivor = @merging.first
+            end
+            @game.start_merge(action.entity, survivor, nonsurvivor, :merge)
+            @merging = nil
+          end
+
+          def mergeable_type(corporation)
+            "Corporations that can merge with #{corporation.name}"
+          end
+
+          def mergeable_candidates(corporation)
```

Comment:
Method called multiple times. Is it worth caching and then clearing in the setup method?

Reply:
Maybe, but 1867 will have the same issue since that's where I stole this from.

Reply:
I'll cache it.